out/blocks/wd_admin_payment_main_form.tpl

```diff
@@ -2,31 +2,93 @@
+      color: red;
+    }
 </style>
 
 <script type="text/javascript">
+  function addClassToElements(elementIds, className) {
+    elementIds.forEach(function(elementId) {
+      var element = document.getElementById(elementId);
+
+      if (element) {
+        element.classList.add(className);
+      }
+    });
+  }
+
+  function removeClassFromElements(elementIds, className) {
+    elementIds.forEach(function(elementId) {
+      var element = document.getElementById(elementId);
+
+      if (element) {
+        // if no class name was passed, remove all classes from the element
+        if (!className) {
+          element.className = '';
+        } else {
+          element.classList.remove(className);
+        }
+      }
+    });
+  }
+
   function testPaymentMethodCredentials() {
-    // the DOM element the test result will be rendered to
-    var resultSpan = document.getElementById('test_credentials_result');
+    var checkSuccess = false;
+
+    var successText = '[{oxmultilang ident="success_credentials"}]';
+    var failureText = '[{oxmultilang ident="error_credentials"}]';
+
+    var successClassName = 'color-success';
+    var failureClassName = 'color-failure';
+
+    // the id of the DOM element the test result will be rendered to
+    var resultSpanId = 'test_credentials_result';
+
+    // the ids of the DOM elements that should be marked red upon test failure
+    var labelConfigApiUrlId = 'labelConfigApiUrl';
+    var labelConfigHttpUserId = 'labelConfigHttpUser';
+    var labelConfigHttpPassId = 'labelConfigHttpPass';
+
+    var resultSpan = document.getElementById(resultSpanId);
+
+    var showCheckResultText = function(success) {
+      resultSpan.innerHTML = success ? successText : failureText;
+      resultSpan.classList.add(success ? successClassName : failureClassName);
+    };
 
     // clear any previous test results
     resultSpan.innerHTML = '';
+    removeClassFromElements([resultSpanId, labelConfigApiUrlId, labelConfigHttpUserId, labelConfigHttpPassId]);
+
+    // build check configuration request
+    var requestUrl = '[{ $oViewConf->getAjaxLink() }]cmpid=container&container=payment_main&fnc=checkPaymentMethodCredentials';
 
     var bodyParams = {
       apiUrl: document.getElementById('configApiUrl').value,
       httpUser: document.getElementById('configHttpUser').value,
       httpPass: document.getElementById('configHttpPass').value
     };
 
-    var successText = '[{oxmultilang ident="success_credentials"}]';
-    var failureText = '[{oxmultilang ident="error_credentials"}]';
-
-    var requestUrl = '[{ $oViewConf->getAjaxLink() }]cmpid=container&container=payment_main&fnc=checkPaymentMethodCredentials';
-
     var paramString = Object.keys(bodyParams).map(function(key) {
       return encodeURIComponent(key) + '=' + encodeURIComponent(bodyParams[key]);
     }).join('&');
 
+    // check if API URL is a root URL, if not there is no need to perform the request
+    var regexCheck = /^https\:\/\/[^\/]+$/;
```

Comment:
This is quite dangerous if they ever decide to change the API URL in the future. E.g. this will fail:
`https://api.wirecard.com/legacy`

Reply:
Yes, this is the desired behaviour.

Reply:
Yes, this is the desired behaviour.